models/amity.py

```diff
@@ -0,0 +1,428 @@
+                        " of Employee ID " + str(new_person.emp_id) + \
+                        "\nthe following living space: " + \
+                        living_choice.name
+                    print spacer
+                else:
+                    print "There are no living spaces in the system."
+                    print "Add a living space using the create_room command " \
+                        "and try again."
+                    print spacer
+                    return
+            else:
+                new_person = Fellow(name)
+                self.success_added_person(new_person, wants_space)
+                self.fellows.append(new_person)
+        self.people.append(new_person)
+
+    def add_person_from_db(self, args):
+        """Add new person from an exsiting database"""
+        print spacer
+        is_fellow = args["is_fellow"]
+        first_name = args["first_name"]
+        last_name = args["last_name"]
+        emp_id = args["emp_id"]
+        wants_space = "No"
+        if is_fellow:
+            new_person = FellowFromDatabase(
+                is_fellow, first_name, last_name, emp_id
+            )
+            self.fellows.append(new_person)
+        else:
+            new_person = StaffFromDatabase(
+                is_fellow, first_name, last_name, emp_id
+            )
+            self.staff.append(new_person)
+        self.success_added_person(new_person, wants_space)
+        self.people.append(new_person)
+
+    def success_added_person(self, new_person, wants_space):
+        """Success message when person has been successfully added"""
+        print "You have successfully added the following person:"
+        print "Name: " + new_person.name + " | Employee ID: " + \
+            str(new_person.emp_id) + \
+            "\nJob Type: " + new_person.job_type + \
+            " | Wants Space?: " + wants_space
+        print spacer
+
+    def reallocate_person(self, args):
+        """(Re)allocate person to (another) room"""
+        print spacer
+        emp_id = int(args["<employee_id>"])
+        for p in self.people:
+            if p.emp_id == emp_id:
+                new_person = p
+        new_room_name = args["<new_room_name>"]
+        for r in self.rooms:
+            if r.name == new_room_name:
+                new_room = r
+
+        """Check that employee ID entered exists"""
+        if emp_id not in [p.emp_id for p in self.people]:
+            print "The employee ID you have entered does not exist."
+            print "Please try again."
+            print spacer
+            return
+
+        """Check that room entered exists"""
+        if new_room_name not in [r.name for r in self.rooms]:
+            print "The room you have entered does not exist."
+            print "Please try again."
+            print spacer
+            return
+
+        """Check that room entered is vacant"""
+        if new_room_name not in [r.name for r in self.vacant_rooms]:
```

Comment:
**Code Quality: Critical**

You would have checked all this in one loop as opposed to breaking it into two checks (room exists and room vacant).